tests/test_other.py

```diff
@@ -11246,7 +11246,16 @@ def test_wasmfs_readfile(self):
     self.set_setting('WASMFS')
     self.do_run_in_out_file_test(test_file('wasmfs/wasmfs_readfile.c'))
 
-  def test_wasmfs_jsfile(self):
+  @parameterized({
+    '': (False, [],),
+    'proxying_backend': (True, ['USE_PTHREADS', 'PROXY_TO_PTHREAD', 'EXIT_RUNTIME'],),
+  })
+  def test_wasmfs_jsfile(self, proxying, args):
+    for setting in args:
+      self.set_setting(setting)
+    if proxying:
+      self.emcc_args = self.emcc_args.copy() + ['-DPROXYING']
+      self.node_args += ['--experimental-wasm-threads', '--experimental-wasm-bulk-memory']
```

Comment:
Grepping the codebase, this would be the first test to hardcode these node flags instead of using `@node_pthreads`. I assume you hardcoded them because that decorator is not conditional, but you wanted it just in one mode?

I think it might be better to have two separate test functions, one decorated and the other not. To avoid code duplication, the second can call the first, or something like that. Another option might be to add a new decorator, `@also_with_node_pthreads` that runs in both modes, but maybe that's overkill?

Reply:
Yep, that's correct. Separating them is the right move here.